docs/tutorial/frontend-tut.md

```diff
@@ -1,94 +1,117 @@
-In the above example, the `map` expression multiplies the values of `avec` and `bvec`.
-`map 1` states that the operation has a *parallelism factor* of 1 which means that the loop iterations are performed sequentially.
 
-`reduce` expressions walk over memories and accumulate a result into a register.
-In the above code snippet, we add together all the elements of `prodvec` and place them in a register named `dot`.
-Since the `reduce` parallelism factor is also 1, the reduction is performed sequentially.
+## Example: sum of squares
+
+Here's a MrXL program that squares and then sums the values of an input array:
+```
+{{#include ../../frontends/mrxl/test/sos.mrxl}}
+```
+
+This short program shows off all of MrXL's features, so let's pick it apart line by line:
+1. We specify an array, `avec`, which will have four integers. The `input` keyword means that an external harness will populate the array.
+2. We specify `sos`, a register. The `output` keyword means that we will populate `sos` in our program.
+3. The `map` operation gets the values of `avec` and raises each to the second power. We stash the result in a new array, `squares`. The number `1` denotes a *parallelism factor* of 1, meaning that the operation is performed sequentially. We will improve this shortly.
+4. The `reduce` operation walks over `squares` and accumulates the result into a register. The parallelism factor is again `1`.
 
 
-## Run a MrXL Program
+## Running our example
 
-Once we have [installed the mrxl command line tool][mrxldocs], we can run MrXL programs using [`fud`][fud].
+Let's run this program.
 
-To provide MrXL program with input values, we use fud's [JSON][json]-based [data format][fud-data].
-Let's try to run this program, which has a parallelism factor of two:
+To begin, [install the MrXL command line tool][mrxldocs-install].
+(TK: I'd kinda like to inline those instructions here, so that there is less jumping around between files.
+I want folks to just install and get back here so we can interpret together, but there's a risk that folks will just begin to follow the other guide and go all the way to Verilog there.
+Also, the MrXL tutorial has a different running example.
+Thoughts on the inline?)
 
+Now change directories to `calyx/frontends/mrxl` and run:
 ```
-{{#include ../../frontends/mrxl/test/add.mrxl}}
+mrxl test/sos.mrxl --data test/sos.mrxl.data --interpret
 ```
-In order to take advantage of the parallelism in the program, the MrXL compiler automatically partitions the input memory `foo` into two different *physical banks*: `foo_b0` and `foo_b1`.
-Therefore, we split up our logical `foo` input of `[1, 2, 3, 4]` into `[1,2]` and `[3,4]`:
+
+Why `42`? Because we populated `avec` with:
 ```json
-{{#include ../../frontends/mrxl/test/add.mrxl.data:2:23}}
+{{#include ../../frontends/mrxl/test/sos.mrxl.data}}
 ```
-Our complete data file similarly splits up the input for `baz`.
+and $0^2 + 1^2 + 4^2 + 5^2 = 42$.
```

Comment:
This doesn't seem to get rendered as math in the resulting doc. Can you try using `$$` instead and see if that works?

Reply:
Huh, weird. Github-flavored Markdown seems okay with it. Will try!

Reply:
Fixed!